### PR TITLE
Implement option to cycle throu the capitalization modes and add option to set spacebar to the spacebar character

### PR DESCRIPTION
**Pull request by lm41**
Added option to select between:

(Spacebar character),
image
(Nothing) and
image
(Current language)

Closes: #2350 
Closes: #419

### CODE REVIEW

**Comment by lm41** [comment]
Converted to draft to add #419

**Review by patrickgold — CHANGES_REQUESTED** [review]
Thanks for implementing those 2 features!

The spacebar label is awesome and the Settings UX for it is also understandable, great work!

As for the cycle through capitalization options, the functionality itself works and is fine, however I am not too fond of the Settings UX for it. In FlorisBoard a switch is mostly used to enable/disable something (sometimes in combination with an enum but still), your switch just changes the behavior of something and changes the summary text, which is a bit confusing. Could you change that to be a 2 option enum? The pref could be titled "Capitalization behavior" or something and the options named accordingly, maybe even with an additional explanation like done in the "Hinted number row" pref (although the detailed explanation may not be necessary if the title and options are well named).

Additionally I have some formatting nitpickings, nothing drastic just some unnecessary whitespace lines and missing trailing commas in param lists (I know that's optional but makes VCS changes look cleaner if consistently used)

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -850,6 +851,9 @@
     <string name="enum__utility_key_action__switch_keyboard_app" comment="Enum value label">Switch keyboard app</string>
     <string name="enum__utility_key_action__dynamic_switch_language_emojis" comment="Enum value label">Dynamic: Switch to emojis / Switch language</string>
 
+    <string name="enum__space_bar_mode__nothing" comment="Enum value label">No label</string>
+    <string name="enum__space_bar_mode__current_language" comment="Enum value label">Current language</string>
+    <string name="enum__space_bar_mode__space_bar_key" comment="Enum value label">␣</string>
```
[the remark this answers is not in the document]
Comment (lm41) [reply app/src/main/res/values/strings.xml:857]: Good to know. I haven't noticed the alphabetical order.

**Review by patrickgold — APPROVED** [review]
Perfect thanks for the changes. Reviewed the code and tested out the changes and everything works great. Merging into master now.

**Pull request merged**